site/src/pages/docs/advanced-spring-boot-integration.mdx

```diff
@@ -0,0 +1,421 @@
+      protocols:
+        - HTTP
+  # Add the below configuration
+  internal-services:
+    include: docs, health, metrics, actuator  # you can add all of the services using `all`
+    port: 8090
+```
+
+The `armeria.internal-services.port` is unnecessary. If the port is not specified or left as 0,
+it will bind the internal service to a random unused port.
+It is ok to set the port of internal service same as one of `armeria.ports`.
+
+<Tip>
+
+  The internal services described below must be configured through the beans described in the documentation.
+  Do not use the method of <type://ServerBuilder> directly using <type://ArmeriaServerConfigurator> or
+  `Consumer<ServerBuilder>` bean. They may be overwritten by the automatic configuration of Armeria.
+
+</Tip>
+
+### Documentation Service
+
+Armeria has its own [documentation service](/docs/server-docservice) which provides useful features.
+For example, it can test RPC protocol in web browser console like Swagger.
+
+Include the service id `docs` to `armeria.internal-services.include`. The `armeria.docs-path` is not necessary.
+The default path of documentation service is `/internal/docs`.
+
+```yaml filename=application.yml
+armeria:
+  internal-services:
+    include: docs
+  docs-path: /internal/docs
+```
+
+You can add the custom configuration to your documentation service via <type://DocServiceConfigurator> bean.
+
+```java filename=ArmeriaConfiguration.java
+@Bean
+public DocServiceConfigurator docServiceConfigurator() {
+    return docServiceBuilder -> docServiceBuilder
+            .exampleRequests(TodoAnnotatedService.class, "create", "{\"id\":\"42\", \"value\":\"foo bar\"}");
+}
+```
+
+### Health Check Service
+
+You can add customize health check operation to health check internal service with <type://HealthChecker> bean.
+
+Include the service id `health` to `armeria-internal-services.include`. The `armeria.health-check-path` is not necessary.
+The default path is `/internal/healthcheck`.
+
+```yaml filename=application.yml
+armeria:
+  internal-services:
+    include: health
+  health-check-path: /internal/healthcheck
+```
+
+Make <type://HealthChecker> bean which implements your heath check business logic. For example, the below code is checking tomcat connector state.
+
+```java filename=ArmeriaConfiguration.java
+@Bean
+public HealthChecker tomcatConnectorHealthChecker(ServletWebServerApplicationContext applicationContext) {
+    final Connector connector = getConnector(applicationContext);
+    return () -> connector.getState().isAvailable();
+}
+```
+
+Similar with documentation service, you can add custom configuration of the service using <type://HealthCheckServiceConfigurator> bean.
+
+```java filename=ArmeriaConfiguration.java
+@Bean
+public HealthCheckServiceConfigurator healthCheckServiceConfigurator() {
+    return healthCheckServiceBuilder -> healthCheckServiceBuilder
+            .updatable(true)
+            .startUnhealthy();
+}
+```
+
+### Metric Service
```

Comment:
Please don't capitalize each word in the heading for consistency with other pages.

```suggestion
### Collecting metrics
```